sql/core/src/main/scala/org/apache/spark/sql/execution/QueryExecution.scala

```diff
@@ -74,12 +76,31 @@ class QueryExecution(
     sparkSession.sessionState.analyzer.executeAndCheck(logical, tracker)
   }
 
+  // SPARK-35378: Eagerly execute Command
+  // Currently, Command returns GenericInternalRow and some Exec casts the output
+  // of child plan to UnsafeRow. If we set Command as the child node or sub query,
+  // then ClassCastException occurs. Such as:
+  // java.lang.ClassCastException
+  // org.apache.spark.sql.catalyst.expressions.GenericInternalRow cannot be cast to
+  // org.apache.spark.sql.catalyst.expressions.UnsafeRow
+  // So eagerly executes Command and converts the output of it to UnsafeRow.
+  lazy val commandExecuted: LogicalPlan = analyzed transformDown {
+    case c: Command if isExecutingCommand == false =>
```

Comment:
This does eagerly execute the commands, but it doesn't solve the cast issue if commands are nested, e.g. you insert into a table with the result of SHOW TABLES.

Even if `isExecutingCommand = true`, we should still eagerly execute the non-root commands, to turn them into `CommandResult` and avoid the cast issue.

Reply:
I think we need code like this
```
if (isExecutingCommand) {
  analyzed.mapChildren(eagerlyExecuteCommands)
} else {
  eagerlyExecuteCommands(analyzed)
}
...
def eagerlyExecuteCommands(p: LogicalPlan) = p.transformDown {
  ...
}
```